### PR TITLE
Add a Plugin Registry

### PR DESCRIPTION
**Pull request by edwarnicke**

### CODE REVIEW

File: utils/registry/registry.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+// must be implemented by each plugin.
+
+package registry
+
+import (
+	"reflect"
+	"sync"
+
+	"github.com/ligato/networkservicemesh/utils/helper"
+
```
Comment (fkautz) [inline utils/registry/registry.go]: Delete this line and resort the deps.

File: utils/registry/registry.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+// Delete - delete p from registry
+func (r *RegistryImpl) Delete(p idempotent.PluginAPI) {
+	r.Lock()
+	defer r.Unlock()
+	for i, value := range r.plugins {
+		vdeps, _ := helper.GetDeps(value)
+		pdeps, _ := helper.GetDeps(p)
+		if reflect.TypeOf(value) == reflect.TypeOf(p) && reflect.DeepEqual(pdeps, vdeps) {
+			r.plugins = append(r.plugins[:i], r.plugins[i+1:]...)
+			return
```
Comment (fkautz) [inline utils/registry/registry.go]: Return is not necessary here, should be removed.

File: utils/registry/registry.go [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+		}
+	}
+}
+
+// Clear is intended only for testing
```
Comment (fkautz) [inline utils/registry/registry.go]: Is there an interface this is implementing? If not, perhaps we can make a test struct that extends RegistryImpl that knows how to clear the plugins slice.

Comment (edwarnicke) [reply utils/registry/registry.go]: fkautz: That is a very good idea!

**Pull request merged**